app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/BrowserNavigationOverlay.kt

```diff
@@ -267,10 +260,67 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         })
     }
 
+    fun observeForToolbar(fragment: Fragment) {
+        // TODO remove this
+        // See comment in fun observeForMegaTile
+        val factory = fragment.context!!.serviceLocator.viewModelFactory
```

Comment:
nit: use view context which can't be null?

Reply:
`Fragment#getView` is nullable though, unless I'm misunderstanding your suggestion.

Reply:
`observeForToolbar` gets called from `createView`, which inflates a view. This view has a context you can pass in instead of using a nullable context from the fragment.